src/components/relative.js

```diff
@@ -131,20 +131,21 @@ class FormattedRelative extends Component {
     this.scheduleNextUpdate(this.props, this.state);
   }
 
-  componentWillReceiveProps({value: nextValue}) {
+  static getDerivedStateFromProps({ value }, { prevValue }) {
     // When the `props.value` date changes, `state.now` needs to be updated,
     // and the next update can be rescheduled.
-    if (!isSameDate(nextValue, this.props.value)) {
-      this.setState({now: this.props.intl.now()});
+    if (!isSameDate(value, prevValue)) {
+      return {now: this.props.intl.now(), prevValue: value};
```

Comment:
The issue is located here, `this.props.intl.now()` needs to be changed to `intl.now()` and `intl` must be deconstructed from the current props being passed to `static getDerivedStateFromProps()`, the same way as `value` is.